### ISSUE TITLE
Ability to create single command applications

### ISSUE THREAD

**Issue by mcampbell508** [laravel-zero/laravel-zero#211]
Hi

I am aware of the following that you can set a default command in `config/commands.php`:

> Laravel Zero will always run the command specified below when no command name is
    | provided. Consider update the default command for single command applications.

The part that resonates to me is:

> You cannot pass arguments to the default command because they are ignored.

It would be great though if you could just create a single command application, without this setting of default. The use case here is to be able to be able to create a single command application, that is able to take arguments and options.

This is much like in the Symfony docs:

https://symfony.com/doc/current/components/console/single_command_tool.html

```php
#!/usr/bin/env php
<?php
require __DIR__.'/vendor/autoload.php';

use Acme\Command\DefaultCommand;
use Symfony\Component\Console\Application;

$application = new Application('echo', '1.0.0');
$command = new DefaultCommand();

$application->add($command);

$application->setDefaultCommand($command->getName(), true);
$application->run();
```

The workaround for me currently is to just build a Symfony command, but then you lose all the sugar of Laravel & Laravel Zero.

Thanks for this tool, I think it’s cool!

Apologies if this has been discussed before, I could not find anything relating.

**Comment by nunomaduro** [comment]
Sounds a good idea.

Do you have a ideia for the code for this ?

**Comment by mcampbell508** [comment]
Hmmm, I can have a think. 

Maybe if I come up with a proposal before putting up any merge request.

Just off the top of my head, maybe this is something we can also define in the `config/commands.php` file. 

If an end user can declare a `single-command` config option in `config/commands.php` and in some way, if this configuration option is found, the application will underneath just serve that single command, along with `--help` being available but discard anything else?

E.g `config/commands.php`

```php
    'single-command' => \App\Console\Commands\SingleCommand::class, 
```

Or some alternatives:

- We could create separate repository specifically for this, designed for single command apps.

I think a configuration option could be simpler and easier to maintain.

What do you think?

**Comment by owenvoke** [comment]
Just looking back through some issues for Laravel Zero. This seems like a really easy issue to fix. 👍

I've created a dev branch which implements this functionality. Basically, you just set your `default` command in the [`config/commands.php`](https://github.com/laravel-zero/laravel-zero/blob/af88934d224b8c1d02568602a4894c8f64957a11/config/commands.php#L16) file. And then would add a `singleCommand` boolean to your `config/app.php` file.
https://github.com/laravel-zero/framework/compare/7.x...owenvoke:feature/single-command-app

Any thoughts?

**Comment by dbpolito** [comment]
@owenvoke this looks good, but i think we would still do unnecessary things loading any other commands, etc.

Maybe we should be a bit more clear on that, like having `single-command` with class path as @mcampbell508  suggested, null by default, with a clear explanation that everything else below will be ignored, and not load unnecessary things here: https://github.com/laravel-zero/framework/blob/stable/src/Kernel.php#L98

**Comment by owenvoke** [comment]
Fair enough, however I personally think it might be good to load other commands in some situations (i.e. if a user wants to run `app:build` to create a Phar, or needs to call a hidden command such as `schedule:run`).

Not sure how best to do this though. 🤔

**Comment by peterjaap** [comment]
Bytheway, Symfony Console 5.1 now includes a `SingleCommandApplication` for this exact purpose: https://symfony.com/blog/new-in-symfony-5-1-single-command-applications

**Comment by owenvoke** [comment]
If we used that class @peterjaap, I guess we could just assume that they don't want to run any hidden/other commands. 🤷

**Comment by peterjaap** [comment]
@owenvoke true, it's not very optimal in development stage.
